packages/client/src/components/form/FetchButton.tsx

```diff
@@ -114,75 +103,81 @@ const StyledPrimaryButton = styled(PrimaryButton)`
     }`
   }}
 `
-class FetchButton extends React.Component<IFullProps, IFetchButtonState> {
-  constructor(props: IFullProps) {
-    super(props)
-    this.state = {
-      loading: false,
-      error: false,
-      isDisconnected: false
-    }
-  }
+const FetchButton = (props: IFullProps) => {
+  const {
+    intl,
+    label,
+    modalTitle,
+    successTitle,
+    errorTitle,
+    queryData,
+    isDisabled,
+    onFetch
+  } = props
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(false)
+  const [success, setSuccess] = useState(false)
+  const [show, setShow] = useState(false)
+  const [isDisconnected, setIsDisconnected] = useState(!isNavigatorOnline())
+  const [networkError, setNetworkError] = useState(false)
 
-  componentDidMount() {
-    this.handleConnectionChange()
-    window.addEventListener('online', this.handleConnectionChange)
-    window.addEventListener('offline', this.handleConnectionChange)
+  const handleConnectionChange = () => {
+    const condition = isNavigatorOnline() ? 'online' : 'offline'
+    if (condition === 'online') {
+      return setIsDisconnected(false)
+    }
+    return setIsDisconnected(true)
   }
 
-  componentWillUnmount() {
-    window.removeEventListener('online', this.handleConnectionChange)
-    window.removeEventListener('offline', this.handleConnectionChange)
-  }
+  useEffect(() => {
+    handleConnectionChange()
+    window.addEventListener('online', handleConnectionChange)
+    window.addEventListener('offline', handleConnectionChange)
 
-  handleConnectionChange = () => {
-    const condition = isNavigatorOnline() ? 'online' : 'offline'
-    if (condition === 'online') {
-      return this.setState({ isDisconnected: false })
+    return () => {
+      window.removeEventListener('online', handleConnectionChange)
+      window.removeEventListener('offline', handleConnectionChange)
     }
+  }, [])
```

Comment:
You can use the `useOnlineStatus` hook here, and update the hook using this eventListenters rather than using interval which is used in the hook